class.jetpack.php

```diff
@@ -4736,6 +4739,26 @@ function authenticate_jetpack( $user, $username, $password ) {
 		return new WP_User( $token_details['user_id'] );
 	}
 
+	// Authenticates requests from Jetpack server to WP API endpoints.
+	// Uses the existing XMLRPC oAuth implementation.
+	function wp_json_authenticate( $error ) {
```

Comment:
I would also call this `wp_rest_authenticate` - the `wp-json` nomenclature is an artifact from v1 of the WP REST API plugin and isn't really used anywhere else.